src/Adapter/Order/CommandHandler/CancelOrderProductHandler.php

```diff
@@ -138,9 +136,7 @@ public function handle(CancelOrderProductCommand $command)
         if (!empty($orderDetails['customizedProductsOrderDetail'])) {
             foreach ($orderDetails['customizedProductsOrderDetail'] as $orderDetail) {
                 $qtyCancelProduct = abs($orderDetails['customizedCancelQuantity'][$orderDetail->id_customization]);
-                if (!$order->deleteCustomization($orderDetail->id_customization, $qtyCancelProduct, $orderDetail)) {
-                    $this->errors[] = $this->translator->trans('An error occurred while attempting to delete product customization.', [], 'Admin.Orderscustomers.Notification') . ' ' . $id_customization;
-                }
+                $this->orderProductRemover->deleteProductFromOrder($order, $orderDetail, $qtyCancelProduct, true);
```

Comment:
You could get rid of the last parameter (since the info is already in the the OrderDetail object)
This way you could even get rid of this loop

Besides I'm worried that calling this method twice (in case there is a customization) will decrement the quantity twice as well (unless the `$orderDetails['productsOrderDetails']` only contains OrderDetail without customization, then they will not overlap)

Reply:
Well this process is called twice today (both in legacy and on 1.7.7.x) , so we are iso legacy. 
But I'll still have a look at what you say, in case I could fix an existing bug 🤔

Reply:
yes `$orderDetails['productsOrderDetails']` only contains OrderDetail without customization.